nemo/utils/exp_manager.py

```diff
@@ -877,10 +877,16 @@ def nemo_topk_check_previous_run(self):
         else:
             models_to_delete = len(best_k_models) - self.save_top_k
         logging.debug(f'Number of models to delete: {models_to_delete}')
+
+        # If EMA enabled, delete the additional EMA weights
+        ema_enabled = self._has_ema_ckpts(best_k_models)
+
         for _ in range(models_to_delete):
```

Comment:
One note here: since all `models_to_delete` only have reg checkpoint path.
If in the folder, there are some EMA-checkpoints that do not have accompanied reg checkpoints (and they are supposed to be remove), these EMA-checkpoint will not be removed.

I think this edge case is fine to leave it as it is because it should never happen in an actual training setup: When we have EMA ckpts saved, the reg ckpts should have always have been created already before.

Reply:
agreed!